pandas/io/common.py

```diff
@@ -635,8 +651,13 @@ def get_handle(
         check_parent_directory(str(handle))
 
     if compression:
-        # compression libraries do not like an explicit text-mode
-        ioargs.mode = ioargs.mode.replace("t", "")
+        if compression != "zstd":
+            # compression libraries do not like an explicit text-mode
+            ioargs.mode = ioargs.mode.replace("t", "")
+        elif compression == "zstd" and "b" not in ioargs.mode:
+            # python-zstandard defaults to text mode, but we always expect
+            # compression libraries to use binary mode.
+            ioargs.mode += "b"
```

Comment:
I tried hard to come up with a case in which this happens but I don't think I understand the entire file opening complexity. Can you show a case where this will be a problem?

Reply:
You are right, this probably shouldn't happen :)